corehq/apps/hqadmin/forms.py

```diff
@@ -114,6 +97,85 @@ def __init__(self, can_toggle_is_staff, *args, **kwargs):
         )
 
 
+class OffboardingUserListForm(forms.Form):
+    csv_email_list = forms.CharField(
+        label="Comma/new-line seperated email addresses",
+        widget=forms.Textarea(),
+        required=False
+    )
+
+    def clean(self):
+        return clean_data(self.cleaned_data, offboarding_list=True)
+
+    def __init__(self, *args, **kwargs):
+        super(OffboardingUserListForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = "form-horizontal"
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.layout = crispy.Layout(
+            'csv_email_list',
+            FormActions(
+                crispy.Submit(
+                    'superuser_management',
+                    'Get Users Not in List'
+                )
+            )
+        )
+
+
+def clean_data(cleaned_data, offboarding_list=False):
+    EMAIL_INDEX = 1
+    csv_email_list = cleaned_data.get('csv_email_list', '')
+    all_users = User.objects.filter(Q(is_superuser=True) | Q(is_staff=True)
+                                    | (Q(is_active=True) & Q(username__endswith='@dimagi.com')))
+    if not csv_email_list:
+        cleaned_data['csv_email_list'] = all_users
+        return cleaned_data
```

Comment:
Does this apply equally to "Grant superuser privileges" and "Get users for offboarding"? My read of this code is that if you submit "Grant superuser privileges" empty, it'll apply your is_superuser / is_staff selection to all dimagi email addresses + everyone currently with is_superuser, which would not be what we want! Can you confirm locally this isn't what would happen?

Reply:
You can't submit a blank email list on the superuser management form so it'll technically never reach this point, but I added a condition requiring `offboarding_list` to be true here and explicitly set `required=true` for the csv_email_list form field (its required by default) for better readability.

Reply:
Great, thank you! It's a bad enough outcome that I think your choice to add these extra guard layers was wise :)